direct/src/dist/locations.py

```diff
@@ -0,0 +1,16 @@
+import sys
+
+if sys.version_info < (3, 10):
```

Comment:
I am not sure this check should be for 3.10, unless we are sure that no Ubuntu/Debian versions exist with a broken version of Python 3.10.  I suggest we do some digging to find out where/when they fixed it.

Reply:
@rdb, had a dig around for this on Debian Salsa. Looking at the change log it was Debian `python3.10 (3.10.2-3)` on Fri, 25 Feb 2022, when they re-introduced patching sysconfig. This does mean that there may be some people with a broken sysconfig on python3.10, but updating their python via APT should solve it. 
Ultimately, it's a judgement call for you to make. 
https://salsa.debian.org/cpython-team/python3/-/blob/master/debian/changelog#L202

Reply:
Instead of a version check, I would prefer that we just attempt to use `distutils` and fallback to `sysconfig` on an `ImportError`.

Reply:
I'm happy with either an ImportError fallback, a version check for for Python 3.12+ or a version check for Python 3.10.3+.

Being certain that we don't break any build configuration weighs more heavily than possibly having a deprecation warning be shown in some configurations.